### ISSUE TITLE
Begin animation from current state

### ISSUE THREAD

**Issue by pronebird** [Microsoft/reactxp#401]
Hi,

Is it possible to continue animation from the current state? I'd like to reverse the animation but I don't see the way to obtain the current interpolated value. Using `containerHeight` that I obtain in `onLayout` seems to be not very precise and produces choppy animation.

I'd be totally cool to continue from the current state if I could omit the `fromValue` just like on iOS, but that doesn't seem to be possible with RX API

```
// create new animated value
const fromValue =  Animated.createValue(this.state.containerHeight);

// whether collapse or expand
const toValue = this.props.height === 'auto' ?
      this.state.contentHeight :
      this.props.height;

// stop previous animation
if(this.state.animation) {
  this.state.animation.stop();
}

// create new animation
const animation = Animated.timing(fromValue, {
  toValue: toValue,
  duration: 2500,
  easing: Animated.Easing.InOut,
});

// save new animation and value
this.setState({
  animatedHeightValue: fromValue,
  animation: animation,
});

// start animating
animation.start();
```

**Comment by erictraut** [comment]
If you simply want to start animating from the last value (whether it was the result of a completed or a "stop"ed animation), you can simply start a new animation and specify the "to" value. There's no need to query the previous value. For example, if the starting value is 0 and the "to" value is 200, you can then start a second animation and specify the "to" value of 0 to animate it in the reverse direction.

**Comment by pronebird** [comment]
@erictraut so I can pass null as first argument to `Animated.timing()`?
```
const animation = Animated.timing(null, {
  toValue: toValue,
  duration: 2500,
  easing: Animated.Easing.InOut,
});
```

**Comment by erictraut** [comment]
Is this on the web? It’s possible that the web implementation always updates the value to the final “toValue” when stop is called. I can check when I’m back at my computer. The problem is that ReactXP uses CSS transitions for web animations, and CSS doesn’t (as far as I know) provide a way to query the value of a partly-completed transition.

**Comment by pronebird** [comment]
It's on the web. Bummer.

As per spec we should be able to access intermediate values:

>Transitions are a presentational effect. The computed value of a property transitions over time from the old value to the new value. Therefore __if a script queries the computed value__ of a property (or other data depending on it) as it is transitioning, __it will see an intermediate value__ that represents the current animated value of the property.

Reference: https://www.w3.org/TR/css-transitions-1/#transitions-

My bet is, `getComputedStyle()` should be able to report the current value back to JavaScript.

**Comment by erictraut** [comment]
I just rewrote the web implementation of the animation APIs. I was able to use your suggested getComputedStyle trick for most cases. There are two cases where this is especially difficult, and I didn't bother implementing these:
1. Interpolated values. (It's difficult because we'd need to reverse-interpolate the value returned by getComputedStyle.)
2. Transforms. (It's difficult because getComputedStyle() returns the transform matrix rather than the individual transform values. We'd need to do a bunch of matrix calculations to pull apart the individual transform values - and do it in an order that's consistent with the specified transforms.)
